### ISSUE TITLE
async support?

### ISSUE THREAD

**Issue by linux-china** [bgpkit/oneio#56]
Now it's some hard to mix oneio with tokio async style.

**Comment by digizeph** [comment]
Agreed! The difficulties is that non of the compression algorithm libraries we use supports async interface. If you have any suggestions, or even better a PR, I'd love to get async support integrated.
